src/main/java/com/minecolonies/coremod/entity/ai/mobs/EntityAIAttackRangedBowBarbarian.java

```diff
@@ -0,0 +1,196 @@
+package com.minecolonies.coremod.entity.ai.mobs;
+
+import net.minecraft.entity.EntityLivingBase;
+import net.minecraft.entity.ai.EntityAIBase;
+import net.minecraft.init.Items;
+import net.minecraft.item.ItemBow;
+import net.minecraft.util.EnumHand;
+
+/**
+ * Class implementing the AI for the Archer Barbarians ranged attack.
+ */
+public class EntityAIAttackRangedBowBarbarian extends EntityAIBase
+{
+    private final AbstractArcherBarbarian entity;
+    private final double                  moveSpeedAmp;
+    private       int                     attackCooldown;
+    private final double                  maxAttackDistance;
+    private int attackTime = -1;
+    private int     seeTime;
+    private boolean strafingClockwise;
+    private boolean strafingBackwards;
+    private              int    strafingTime         = -1;
+    private final        int    ONE_MINUTE           = 60;
+    private final        int    TWENTY_SECONDS       = 20;
+    private final        double ONE_QUARTER_BLOCK    = 0.25D;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Make this final field static too. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1170)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this field "ONE_QUARTER_BLOCK" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00116)